### PR TITLE
[2679] Implement reactions info feature

### PR DESCRIPTION
**Pull request by bychkovdmitry**
https://github.com/GetStream/stream-chat-android/issues/2679

### 🎯 Goal

Add the `ReactionsInfo` component that shows a list of reactions for a particular message.

### 🛠 Implementation details

Ended up with a single `SelectedMessageState` and 3 overlays:
- MessageOverlay - an abstract message overlay with 3 slots
  - MessageOptionsOverlay - the message options overlay with action list footer
  - MessageReactionsOverlay  - the message reaction overlay with reaction list footer

### ☑️ Checklist

- [x] I have signed the [Stream CLA](https://docs.google.com/forms/d/e/1FAIpQLScFKsKkAJI7mhCr7K9rEIOpqIDThrWxuvxnwUq2XkHyG154vQ/viewform) (required)
- [x] PR targets the `develop` branch
- [x] Changelog is updated with client-facing changes
- [ ] New code is covered by unit tests
- [x] Comparison screenshots added for visual changes
- [ ] Affected documentation updated (docusaurus, tutorial, CMS (task created))
- [x] Reviewers added

### CODE REVIEW

File: stream-chat-android-compose/src/main/java/io/getstream/chat/android/compose/state/messages/MessagesState.kt [hunk 1]
```diff
@@ -22,7 +21,7 @@ public data class MessagesState(
     val isLoadingMore: Boolean = false,
     val endOfMessages: Boolean = false,
     val messageItems: List<MessageListItemState> = emptyList(),
-    val selectedMessage: Message? = null,
+    val selectedMessageState: SelectedMessageState? = null,
```
Comment (bychkovdmitry) [inline stream-chat-android-compose/src/main/java/io/getstream/chat/android/compose/state/messages/MessagesState.kt:24]: Had to make this a sealed class. Having a separate `selectedMessageReactions: Message?` turned our to spawn a lot of boilerplate.

File: stream-chat-android-compose/src/main/java/io/getstream/chat/android/compose/ui/messages/MessagesScreen.kt [hunk 2]
```diff
@@ -169,17 +172,34 @@ public fun MessagesScreen(
...
+                MessageReactionsOverlay(
+                    message = selectedMessage,
+                    currentUser = user,
+                    onMessageAction = { action ->
+                        composerViewModel.performMessageAction(action)
+                        listViewModel.performMessageAction(action)
+                    },
+                    onDismiss = { listViewModel.removeOverlay() }
+                )
+            }
```
Comment (bychkovdmitry) [inline stream-chat-android-compose/src/main/java/io/getstream/chat/android/compose/ui/messages/MessagesScreen.kt:202]: A bit of duplication here.

**Review by filbabic — COMMENTED** [review]
Left some comments where we could try and polish out a few details, but this looks really good! :]

**Review by skydoves — COMMENTED** [review]
Looks really good! I left a couple of comments. 😄

File: stream-chat-android-compose/src/main/java/io/getstream/chat/android/compose/ui/messages/overlay/MessageReactionsOverlay.kt [hunk 3]
```diff
@@ -0,0 +1,222 @@
...
+
+            Spacer(modifier = Modifier.height(16.dp))
+
+            val columns = reactionCount.coerceAtMost(maxColumns)
+            val reactionItemWidth = ChatTheme.dimens.messageReactionItemWidth
+            val reactionGridWidth = reactionItemWidth * columns
+
+            LazyVerticalGrid(
+                modifier = Modifier.width(reactionGridWidth),
+                cells = GridCells.Fixed(reactionCount.coerceAtMost(columns))
```
[the remark this answers is not in the document]
Comment (bychkovdmitry) [reply stream-chat-android-compose/src/main/java/io/getstream/chat/android/compose/ui/messages/overlay/MessageReactionsOverlay.kt:154]: Yeah, indeed, I will check what I can do to satisfy all the requirements:
- 1 reaction - centered
- 2 reactions - centered
...
- N reactions, where N is the max column count which fits into the card width

File: stream-chat-android-compose/src/main/java/io/getstream/chat/android/compose/ui/messages/overlay/MessageReactionsOverlay.kt [hunk 4]
```diff
@@ -0,0 +1,222 @@
...
+ * @param currentUser The currently logged in user.
+ * @param modifier Modifier for styling.
+ */
+@Composable
+public fun UserReactionItem(
+    reaction: Reaction,
+    currentUser: User?,
+    modifier: Modifier = Modifier,
+) {
+    val user = reaction.user!!
```
[the remark this answers is not in the document]
Comment (bychkovdmitry) [reply stream-chat-android-compose/src/main/java/io/getstream/chat/android/compose/ui/messages/overlay/MessageReactionsOverlay.kt:183]: Yeah, will do

File: stream-chat-android-compose/src/main/java/io/getstream/chat/android/compose/ui/messages/overlay/MessageReactionsOverlay.kt [hunk 5]
```diff
@@ -0,0 +1,222 @@
...
+ * Represent a section with a list of reactions left for the message.
+ *
+ * @param message The message that contains the reactions.
+ * @param currentUser The currently logged in user.
+ * @param modifier Modifier for styling.
+ * @param maxColumns The maximum number of columns in the user reactions grid.
+ */
+@OptIn(ExperimentalFoundationApi::class)
+@Composable
+public fun ReactionsInfo(
```
[the remark this answers is not in the document]
Comment (bychkovdmitry) [reply stream-chat-android-compose/src/main/java/io/getstream/chat/android/compose/ui/messages/overlay/MessageReactionsOverlay.kt:112]: Lets merge you PR first

**Pull request closed without merging**